ironfish-contracts/contracts/Ironfish.sol

```diff
@@ -4,19 +4,38 @@ pragma solidity ^0.8.18;
 contract Ironfish {
     event Shield(
         bytes32 ironfishAddress,
-        bytes32 ironfishAssetId,
+        uint tokenId,
         address contractAddress,
         uint amount
     );
 
+    event UnShield(
+        address contractAddress,
+        uint tokenId,
+        uint amount
+    );
+
     constructor() {}
 
     function shield(
         bytes32 ironfishAddress,
-        bytes32 ironfishAssetId,
+        uint tokenId,
         uint amount
     ) public {
         // msg.sender is the contract address of the ERC20 token
-        emit Shield(ironfishAddress, ironfishAssetId, msg.sender, amount);
+        emit Shield(ironfishAddress, tokenId, msg.sender, amount);
+    }
+
```

Comment:
If we are making Iron the native currency, rather than ERC20, then instead we would just be looking for sends to the global contract address and get the amount from that.